integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -572,3 +577,148 @@ fn assert_ft_balance(
     let balance = std::str::from_utf8(&response.result).expect("invalid UTF8");
     assert_eq!(format!("\"{expected_balance}\""), balance);
 }
+
+/// Test account creation scenarios with meta transactions.
+///
+/// Named accounts aren't the primary use case for meta transactions but still
+/// worth a test case.
+#[test]
+fn meta_tx_create_named_account() {
+    let relayer = bob_account();
+    let sender = alice_account();
+    let new_account = eve_dot_alice_account();
+    let node = RuntimeNode::new(&relayer);
+
+    let fee_helper = fee_helper(&node);
+    let amount = NEAR_BASE;
+
+    let public_key = PublicKey::from_seed(KeyType::ED25519, &new_account);
+
+    // That's the minium to create a (useful) account.
+    let actions = vec![
+        Action::CreateAccount(CreateAccountAction {}),
+        Action::Transfer(TransferAction { deposit: amount }),
+        Action::AddKey(AddKeyAction { public_key, access_key: AccessKey::full_access() }),
+    ];
+
+    // Check account doesn't exist, yet
+    node.view_account(&new_account).expect_err("account already exists");
+
+    let tx_cost = fee_helper.create_account_transfer_full_key_cost();
+    check_meta_tx_no_fn_call(&node, actions, tx_cost, amount, sender, relayer, new_account.clone());
+
+    // Check account exists
+    node.view_account(&new_account).expect("failed looking up account");
+}
+
+/// Try creating an implicit account with `CreateAction` which is not allowed in
+/// or outside meta transactions and must fail with `OnlyImplicitAccountCreationAllowed`.
```

Comment:
Observation: isn’t this error variant a misnomer? We’re saying that you can only create implicit accounts where the expectation really is that you can only create _named_ accounts…?

Reply:
Yeah, good point, very confusing.

I think the naming wants to say, "you are trying to create a 64 char length account, this is only allowed in the fancy new implicit account creation flow", where "fancy and new" refers back to 2020 and this NEP that introduced the feature: https://github.com/near/NEPs/pull/71

Before that, named accounts were the only type of account.

Reply:
As it is confusing for all of us - can we rename error to smth like `OnlyNamedAccountCreationAllowed` in separate PR? If this is part of protocol, we can try renaming enum name but leave conversion to string as is.

Reply:
Agreed. I believe it's not a protocol change to rename this, the borsh representation should be fine and I don't think we store errors as strings anywhere it matters. But the RPC consumers might care, we have had unfortunate surprises around that in the past.

For now I filed this: https://github.com/near/nearcore/issues/8598

Reply:
And I added a TODO in the code comment